components/organisation-unit-tree/src/organisation-unit-node/use-org-children.js

```diff
@@ -0,0 +1,56 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { useMemo, useEffect } from 'react'
+import { sortNodeChildrenAlphabetically } from '../helpers/index.js'
+
+const ORG_DATA_QUERY = {
+    orgUnit: {
+        resource: `organisationUnits`,
+        id: ({ id }) => id,
+        params: {
+            fields: 'children[id,path,displayName]',
+        },
+    },
+}
+
+/**
+ * @param {string[]} ids
+ * @param {Object} options
+ * @param {string} options.displayName
+ * @param {boolean} [options.withChildren]
+ * @returns {Object}
+ */
+export const useOrgChildren = ({
+    node,
+    suppressAlphabeticalSorting,
+    onComplete,
+}) => {
+    const { called, loading, error, data } = useDataQuery(ORG_DATA_QUERY, {
+        variables: { id: node.id },
+    })
+
+    const orgChildren = useMemo(() => {
+        if (!data) {
+            return undefined
+        }
+
+        const { orgUnit } = data
+
+        // undefined or zero
+        if (!node.children) {
+            return []
+        }
+
+        return suppressAlphabeticalSorting
+            ? orgUnit.children
+            : sortNodeChildrenAlphabetically(orgUnit.children)
+    }, [data, suppressAlphabeticalSorting])
+
+    useEffect(() => {
+        if (onComplete && orgChildren) {
+            // For backwards compatibility: Pass entire node incl. children
+            onComplete({ ...node, children: orgChildren })
```

Comment:
It's possible that this callback will be called more often than just when data and suppressAlphabeticalSorting change as useMemo does not guarantee to be referentially stable. Is that ok here?

Reply:
Fixed